kernel/generic/src/mm/as.c

```diff
@@ -1398,16 +1390,14 @@ errno_t as_area_change_flags(as_t *as, unsigned int flags, uintptr_t address)
 	mutex_unlock(&area->sh_info->lock);
 
 	/*
-	 * Compute total number of used pages in the used_space B+tree
+	 * Compute total number of used pages
 	 */
 	size_t used_pages = 0;
 
-	list_foreach(area->used_space.leaf_list, leaf_link, btree_node_t,
-	    node) {
-		btree_key_t i;
-
-		for (i = 0; i < node->keys; i++)
-			used_pages += (size_t) node->value[i];
+	used_space_ival_t *ival = used_space_first(&area->used_space);
+	while (ival != NULL) {
+		used_pages += ival->count;
```

Comment:
Is the computed value different from the one stored in area->used_space.pages?

Reply:
No, it's the same. I will change it to simply use  used_space.pages.